packages/jellyfish-transaction/src/script/dftx/dftx_pool.ts

```diff
@@ -41,13 +39,18 @@ export class CPoolSwap extends ComposableBuffer<PoolSwap> {
       ComposableBuffer.single<Script>(() => ps.toScript, v => ps.toScript = v, v => new CScript(v)),
       ComposableBuffer.varUInt(() => ps.toTokenId, v => ps.toTokenId = v),
       {
-        fromBuffer (buffer: SmartBuffer) {
+        fromBuffer: (buffer: SmartBuffer): void => {
           const integer = readBigNumberUInt64(buffer)
           const fraction = readBigNumberUInt64(buffer)
-          ps.maxPrice = { integer, fraction }
+          ps.maxPrice = integer.plus(fraction.dividedBy(ONE_HUNDRED_MILLION))
         },
-        toBuffer (buffer: SmartBuffer) {
-          const { integer, fraction } = ps.maxPrice
+        toBuffer: (buffer: SmartBuffer): void => {
+          const n = ps.maxPrice.multipliedBy(ONE_HUNDRED_MILLION)
+          const fraction = n.mod(ONE_HUNDRED_MILLION)
+          if (fraction > new BigNumber('99999999')) {
```

Comment:
```suggestion
          if (fraction.gt(new BigNumber('99999999'))) {
```
the `>` should not be working here

Reply:
Indeed. Weirdly enough it worked on the tests but I will fix it

Reply:
hmm.. i'm quite out of date on `bignumber` as its now support arithmetic operator

Reply:
Still better with the method as it is more coherent with the codebase. 
I commited the change